commands/build.go

```diff
@@ -120,10 +121,13 @@ func (o *buildOptions) toControllerOptions() (*controllerapi.BuildOptions, error
 		ExportLoad:     o.exportLoad,
 	}
 
-	// TODO: extract env var parsing to a method easily usable by library consumers
-	if v := os.Getenv("SOURCE_DATE_EPOCH"); v != "" {
-		if _, ok := opts.BuildArgs["SOURCE_DATE_EPOCH"]; !ok {
-			opts.BuildArgs["SOURCE_DATE_EPOCH"] = v
+	if _, ok := opts.BuildArgs[epoch.SourceDateEpochEnv]; !ok {
+		v, err := epoch.SourceDateEpoch()
+		if err != nil {
+			return nil, err
+		}
+		if v != nil {
+			opts.BuildArgs[epoch.SourceDateEpochEnv] = strconv.FormatInt(v.Unix(), 10)
```

Comment:
I'm not sure parsing string to time and back to string for this purpose is ideal.

Reply:
I could still use the parsing, and just get the env-var separately; let me check for a bit

Reply:
hm.. no no utility on its own, so if could be;

```go
opts.BuildArgs[epoch.SourceDateEpochEnv] = os.Getenv(epoch.SourceDateEpochEnv)
```

which also looks a bit odd, as we're setting a value that's not the `v != nil` that we check 😞